src/promptflow/promptflow/_core/tool.py

```diff
@@ -39,27 +39,46 @@ def active_instance(cls) -> Optional["ToolInvoker"]:
         return cls._active_tool_invoker
 
 
-def tool(f: Callable) -> Callable:
+def tool(
+    func=None,
+    *,
+    name: str = None,
+    description: str = None,
+    type: str = None,
+) -> Callable:
     """Decorator for tool functions. The decorated function will be registered as a tool and can be used in a flow.
 
-    :param f: The tool function.
-    :type f: Callable
+    :param name: The tool name.
+    :type name: str
+    :param description: The tool description.
+    :type description: str
+    :param type: The tool type.
+    :type type: str
     :return: The decorated function.
     :rtype: Callable
     """
 
-    @functools.wraps(f)
-    def new_f(*args, **kwargs):
-        tool_invoker = ToolInvoker.active_instance()
-        # If there is no active tool invoker for tracing or other purposes, just call the function.
-        if tool_invoker is None:
-            return f(*args, **kwargs)
-        return tool_invoker.invoke_tool(f, *args, **kwargs)
-
-    new_f.__original_function = f
-    f.__wrapped_function = new_f
-    new_f.__tool = None  # This will be set when generating the tool definition.
-    return new_f
+    def tool_decorator(func: Callable) -> Callable:
+        @functools.wraps(func)
+        def new_f(*args, **kwargs):
+            tool_invoker = ToolInvoker.active_instance()
+            # If there is no active tool invoker for tracing or other purposes, just call the function.
+            if tool_invoker is None:
+                return func(*args, **kwargs)
+            return tool_invoker.invoke_tool(func, *args, **kwargs)
+
+        new_f.__original_function = func
+        func.__wrapped_function = new_f
+        new_f.__tool = None  # This will be set when generating the tool definition.
+        new_f.__name = name
+        new_f.__description = description
+        new_f.__type = type
+        return new_f
+
+    # enable use decorator without "()" if all arguments are default values
```

Comment:
May need test case to guard this behavior

Reply:
There has been added to the test case that use decorator without bracket.
![image](https://github.com/microsoft/promptflow/assets/17938940/24252434-791f-4ed1-a88c-8cfe6c85beeb)